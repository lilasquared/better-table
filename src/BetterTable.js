import React from 'react'
import PropTypes from 'prop-types'

import Table from './Table'

const BetterTable = props => (
  <Table 
    { ...React.Children.only(props.children).props }
    hasFixedHeader={props.hasFixedHeader}
  />
)

BetterTable.propTypes = {
  children: (props, propName, componentName) => {
    const prop = props[propName]
    if (React.Children.count(prop) !== 1 || prop.type !== 'table') {
      return new Error(`\`${componentName}\` should have a single child of type \`table\``)
    }
  },
  hasFixedHeader: PropTypes.bool
}

BetterTable.defaultProps = {
  hasFixedHeader: false
}

export default BetterTable