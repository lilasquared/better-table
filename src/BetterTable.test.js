import React from 'react';
import ReactDOM from 'react-dom';
import BetterTable from './BetterTable';

describe('<BetterTable />', () => {

  it('throws when it does not have a child', () => {
    const div = document.createElement('div');
    expect(() => ReactDOM.render(<BetterTable />, div))
      .toThrowErrorMatchingSnapshot();
  })
  
  it('throws when it has more than one child', () => {
    const div = document.createElement('div');
    expect(() => ReactDOM.render(
      <BetterTable>
        <div />
        <div />
      </BetterTable>, div)).toThrowErrorMatchingSnapshot();
  })

  it('renders when it has a single table element as a child', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BetterTable>
        <table></table>
      </BetterTable>, div);
  });
});