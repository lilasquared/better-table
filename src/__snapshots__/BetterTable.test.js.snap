// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BetterTable /> throws when it does not have a child 1`] = `"React.Children.only expected to receive a single React element child."`;

exports[`<BetterTable /> throws when it has more than one child 1`] = `"React.Children.only expected to receive a single React element child."`;
